### DISCUSSION TITLE
Chapter Summaries for the books

### DISCUSSION THREAD

**Discussion by danielschuster-muc** [danielschuster-muc/potter-db#170] [Ideas]
(Update Sept. 2023)
We are still missing an important part of our API: chapter summaries for the Harry Potter Books. 
Use this discussion to discuss and engage with this essential component. 

Some infos:
- @kat-kan worked on this a while ago. Those chapter summaries are good references on the length and expected level of detail. 
- See #741 on how to contribute

**Comment by kat-kan** [comment]
Hi @danielschuster-muc I am a huge HP fan and I would love to work on that!

**Reply by danielschuster-muc** [reply]
Sure, feel free to do so. Thank you for your help!

**Reply by kat-kan** [reply]
Is it okay to create one PR per a book?

**Reply by danielschuster-muc** [reply]
One PR for all books would be better.

**Reply by kat-kan** [reply]
Ok then but it'll take some time :)

**Reply by danielschuster-muc** [reply]
Sure, no problem.
Just make sure to write own summaries and not to copy them from somewhere else.

**Reply by kat-kan** [reply]
Ofc :) Can you please advice how long each summary should be? I am not sure what level of detail you expect

**Reply by danielschuster-muc** [reply]
Mmh, good question. It should definitely not be too long, just a short summary of what happens in each chapter. Maybe 3 or 4 sentences per chapter? Or what would you say?

**Reply by danielschuster-muc** [reply]
After you have finished the chapters of the first book, you can make a pr and set it on draft, then I can see whether you are on right track. Or even after the first few chapters.

**Reply by kat-kan** [reply]
I wanted to suggest the very same thing, I'll create a PR with some chapters. I'll make it short up to 5 sentences.

**Reply by danielschuster-muc** [reply]
How's your progress? I know it's a huge task and will take much time, but maybe you can give me a short status update. Do you need any further help or tips?

**Reply by kat-kan** [reply]
Hello @danielschuster-muc 
I need to admit this is not an easy task :) So far, I wrote a draft for a few chapters (HP 1st book). TBH, I'm struggling with the level of detail that I should contain while still trying to make it short. 

This is a sample of my work:
Chapter 2 - The Vanishing Glass

Harry Potter is already 11 years old, still living with the Dursleys family. They give a roof over his head, but overall they treat him more like a help than a family member. On the day of Dudley’s birthday, the Dursleys want to visit the zoo and due to an unfortunate coincidence, they have to bring along Harry. The boy gets drawn into a conversation with the boa constrictor. When Dudley starts teasing Harry, the glass in front of the boa disappears. The snake escapes its terrarium, thanking Harry for the help, and spreads panic across visitors, Dursleys included. 

Chapter 3 - The Letters From No One

It’s a summer holiday before Harry and Dudley will go to different schools - Harry’s local school and Dudley’s expensive private school. One morning during breakfast, the mail arrives at Dursley’s home. Unexpectedly, there is a first-ever letter to Harry and on top of that, a strange-looking one. Dursleys don’t allow Harry to read his letter and the next day more letters get into the house by all means, no matter how hard Dursleys try to destroy all of them. Uncle Vernon decides to run away from the problem and takes the whole family to a hotel, but the mail is following them. Finally, they end up in the old cabin on the island which can be accessed only by a boat, hoping they get rid of the letters. Harry is counting the seconds till his 11 birthday and suddenly someone is hammering on the cabin’s door.

I'll be grateful for the feedback.

**Reply by danielschuster-muc** [reply]
First of all thank you for your huge effort! It's indeed a very big task and definitely not easy, but I would say you are on a good way to go. 
I think the length and content is good. 4-5 sounds like a good metric for each chapter, not too long, not too short 👍!

Just one thing I noticed about your 2 samples is that in chapter 2 you say that Harry is already 11, but he's counting the seconds until his 11th birthday? That's a bit confusion. But that's just about the content. 

Anyways, I definitely can say you're on the right track! That's exactly what I imagined for the chapters.

**Reply by danielschuster-muc** [reply]
Hi @kat-kan. How is it going with the chapter summaries? I know it's an enormous task and requires a lot of work. Therefore I was wondering whether you need any further help. I was also thinking of splitting it into smaller tasks per book? So maybe others can also work on writing the summaries. No worries, your work is much appreciated, but by splitting it we will probably make faster progress. Or what do you think?

**Reply by kat-kan** [reply]
Hi @danielschuster-muc this is is a great idea. I cannot commit as much time as I would like to :( I am still in the first book.

**Reply by danielschuster-muc** [reply]
Yep, that’s totally fine! Just out of curiosity, at which chapter are you currently?

**Reply by danielschuster-muc** [reply]
@kat-kan as you where the first one who started to work on the chapter summaries I wanted to ask you if you are still interested in working on those, maybe for Hacktoberfest 2023?

**Reply by kat-kan** [reply]
@danielschuster-muc sure thing, I'd be happy to finish my work however honestly the scope is too overwhelming...I can commit to finishing the first book as I have 1/3 chapters ready

**Reply by danielschuster-muc** [reply]
would be amazing if you could commit to finishing the first book! thank you
